src/proxysql.rs

```diff
@@ -0,0 +1,249 @@
+        let datetime_now: DateTime<Local> = Local::now();
+        let date_formatted = datetime_now.format("%Y-%m-%d %H:%M:%S");
+        if self.warmup_time_s > 0 {
+            self.conn.query_drop(format!("INSERT INTO mysql_query_rules (username, mirror_hostgroup, active, digest, apply, comment) VALUES ('{}', {}, 1, '{}', 1, '{}: {}')", query.get_user(), self.readyset_hostgroup, query.get_digest(), MIRROR_QUERY_TOKEN, date_formatted)).expect("Failed to insert into mysql_query_rules");
+            messages::print_info("Inserted warm-up rule");
+        } else {
+            self.conn.query_drop(format!("INSERT INTO mysql_query_rules (username, destination_hostgroup, active, digest, apply, comment) VALUES ('{}', {}, 1, '{}', 1, '{}: {}')", query.get_user(), self.readyset_hostgroup, query.get_digest(), DESTINATION_QUERY_TOKEN, date_formatted)).expect("Failed to insert into mysql_query_rules");
+            messages::print_info("Inserted destination rule");
+        }
+        Ok(true)
+    }
+
+    pub fn load_query_rules(&mut self) -> Result<bool, mysql::Error> {
+        self.conn
+            .query_drop("LOAD MYSQL QUERY RULES TO RUNTIME")
+            .expect("Failed to load query rules");
+        Ok(true)
+    }
+    pub fn save_query_rules(&mut self) -> Result<bool, mysql::Error> {
+        self.conn
+            .query_drop("SAVE MYSQL QUERY RULES TO DISK")
+            .expect("Failed to load query rules");
+        Ok(true)
+    }
+
+    /// This function is used to check the current list of queries routed to Readyset.
+    ///
+    /// # Arguments
+    /// * `conn` - A reference to a connection to ProxySQL.
+    ///
+    /// # Returns
+    /// A vector of tuples containing the digest_text, digest, and schemaname of the queries that are currently routed to ReadySet.
+    pub fn find_queries_routed_to_readyset(&mut self) -> Vec<String> {
+        let rows: Vec<String> = self
+            .conn
+            .query(format!(
+            "SELECT digest FROM mysql_query_rules WHERE comment LIKE '{}%' OR comment LIKE '{}%'",
+            MIRROR_QUERY_TOKEN, DESTINATION_QUERY_TOKEN
+        ))
+            .expect("Failed to find queries routed to ReadySet");
+        rows
+    }
+
+    /// This function is used to check if any mirror query rule needs to be changed to destination.
+    ///
+    /// # Returns
+    ///
+    /// A boolean indicating if any mirror query rule was changed to destination.
+    pub fn adjust_mirror_rules(&mut self) -> Result<bool, mysql::Error> {
+        let mut updated_rules = false;
+        let datetime_now: DateTime<Local> = Local::now();
+        let tz = datetime_now.format("%z").to_string();
+        let date_formatted = datetime_now.format("%Y-%m-%d %H:%M:%S");
+        let rows: Vec<(u16, String)> = self.conn.query(format!("SELECT rule_id, comment FROM mysql_query_rules WHERE comment LIKE '{}: ____-__-__ __:__:__';", MIRROR_QUERY_TOKEN)).expect("Failed to select mirror rules");
```

Comment:
What's going on here with these underscores?

Reply:
This is the LIKE syntax for MySQL, `_` one char, versus `%` multiple chars. Btw, this is what is in the product now, this code has only moved.

Reply:
Hah, somehow I've never run into this syntax yet.